frame/executive/src/lib.rs

```diff
@@ -222,23 +224,32 @@ where
 	OriginOf<Block::Extrinsic, Context>: From<Option<System::AccountId>>,
 	UnsignedValidator: ValidateUnsigned<Call = CallOf<Block::Extrinsic, Context>>,
 {
-	/// Execute all `OnRuntimeUpgrade` of this runtime, and return the aggregate weight.
-	pub fn execute_on_runtime_upgrade() -> frame_support::weights::Weight {
-		<(COnRuntimeUpgrade, AllPalletsWithSystem) as OnRuntimeUpgrade>::on_runtime_upgrade()
-	}
-
 	/// Execute given block, but don't do any of the `final_checks`.
 	///
 	/// Should only be used for testing.
-	#[cfg(feature = "try-runtime")]
-	pub fn execute_block_no_check(block: Block) -> frame_support::weights::Weight {
+	pub fn try_execute_block(
+		block: Block,
+		check_state_root: bool,
+		sanity_checks: frame_try_runtime::SanityCheckTargets,
+	) -> frame_support::weights::Weight {
 		Self::initialize_block(block.header());
 		Self::initial_checks(&block);
 
 		let (header, extrinsics) = block.deconstruct();
 
 		Self::execute_extrinsics_with_book_keeping(extrinsics, *header.number());
 
+		// run the sanity-checks of all pallets.
+		<AllPalletsWithSystem as frame_support::traits::SanityCheck<System::BlockNumber>>::sanity_check(
+			*header.number(),
+			sanity_checks,
+		)
+		.map_err(|e| {
+			sp_runtime::print("failure:");
+			sp_runtime::print(e);
+		})
+		.expect("sanity-checks should not fail");
+
 		// do some of the checks that would normally happen in `final_checks`, but definitely skip
```

Comment:
Looks like to me that you are doing nearly all checks from `final_checks`, but copy paste.  
The State root is also checked?!

Reply:
the state root is checked IFF the `check_state_root: bool` is set. Updated the doc to reflect this.